.github/workflows/ci.yml

```
@@ -0,0 +1,49 @@
+name: Exawind-CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  Formatting:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.13
+      with:
+        source: './app ./src'
+        exclude: '.'
+        extensions: 'H,h,cpp'
+        clangFormatVersion: 13
+  CPU:
+    needs: Formatting
+    runs-on: ubuntu-latest
+    container:
+      image: ecpe4s/exawind-snapshot
+      env:
+         SPACK_MANAGER: /spack-manager
+         E4S_MACHINE: true
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{github.token}}
+      - name: Clone
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Tests
+        working-directory: /spack-manager/environments
+        run: |
+          /bin/bash -c "\
+            mkdir ci-env && \
+            cd ci-env && \
+            ln -s ${{github.workspace}} exawind && \
+            source ${SPACK_MANAGER}/start.sh && \
+            quick-develop -s exawind@master && \
```

```suggestion
echo ${{github.workspace}} && \
source ${SPACK_MANAGER}/start.sh && \
quick-create -s exawind@master && \
spack manager develop --path ${{github.workspace}} --no-clone exawind@master && \
spack env activate -d . && \
spack manager external --latest && \
```
Okay getting even more granular...